machine-learning/solutions/seismic_salt_detection/trainer/model.py

```diff
@@ -50,74 +48,67 @@ def forward_key_to_export(estimator):
+                estimator_spec.export_outputs[
                     ekey
                 ] = tf.estimator.export.PredictOutput(
-                    estimatorSpec.predictions)
-        return estimatorSpec
+                    estimator_spec.predictions)
+        return estimator_spec
     return tf.estimator.Estimator(model_fn=model_fn2, config=config)
 
-def create_regressor(config, parameters):
-    """Creates a DNN regressor.
 
-    Defines features and builds an 'Estimator' with them.
+def _estimator_metrics(labels, predictions):
+    """Creates metrics for Estimator.
+
+    Metrics defined here can be used to evaluate the model (on evaluation
+    data) and also can be used to maximize or minimize their values during
+    hyper-parameter tunning.
 
     Args:
-        config: `RunConfig` object to configure the runtime of the `Estimator`.
-        parameters: Parameters passed to the job.
+        labels: Evaluation true labels.
+        predictions: Evaluation model predictions.
 
     Returns:
-        A configured and ready to use `tf.estimator.DNNRegressor`
+        A dictionary with the evaluation metrics
     """
+    pred_logistic = predictions['logistic']
+    pred_class = predictions['class_ids']
+    return {
+        'accuracy': tf.metrics.accuracy(labels, pred_class),
+        'auc': tf.metrics.auc(labels, pred_logistic),
+        'auc_pr': tf.metrics.auc(labels, pred_logistic, curve='PR'),
+        'precision': tf.metrics.precision(labels, pred_class),
+        'recall': tf.metrics.recall(labels, pred_class)}
 
-    # Mean and Standard Deviation Constants for normalization.
-    with file_io.FileIO(parameters.mean_path, mode='r') as f:
-        mean = pickle.load(f)
-    with file_io.FileIO(parameters.std_path, mode='r') as f:
-        std = pickle.load(f)
-
-    # Columns to be used as features.
-    hour = tf.feature_column.categorical_column_with_identity(
-        'hour', num_buckets=24)
-    hour = tf.feature_column.embedding_column(
-        hour, dimension=parameters.hour_embedding)
 
-    day = tf.feature_column.categorical_column_with_identity(
-        'day', num_buckets=7)
-    day = tf.feature_column.embedding_column(
-        day, dimension=parameters.day_embedding)
+def create_classifier(config, parameters):
+    """Creates a DNN classifier.
 
-    weather = [tf.feature_column.numeric_column(
-        'weather' + str(i),
-        normalizer_fn=(lambda x, i = i: (x - mean[i]) / std[i])
-    ) for i in range(constants.WEATHER_SIZE)]
+    Defines features and builds an 'Estimator' with them.
 
-    distribution = [tf.feature_column.numeric_column(
-        'distribution' + str(i)
-    ) for i in range(constants.DISTRIBUTION_SIZE)]
+    Args:
+        config: `RunConfig` object to configure the runtime of the `Estimator`.
+        parameters: Parameters passed to the job.
 
-    feature_cols = [hour, day] + weather + distribution
+    Returns:
+        A configured and ready to use `tf.estimator.DNNClassifier`
```

Comment:
@wes-turner
wes-turner 6 days ago  Collaborator
What makes an estimator "configured"? The docstrings in this module haven't, in general, been very helpful to me. They feel a little like they were written so that the file would have docstrings.

If it's because I don't have the TensorFlow experience that most people will have (and this file is assuming), that's ok, too. In that case, I think I recommend just using the single-line short form.

  @alan-krumholz
alan-krumholz 5 days ago  Owner
configured = "estimator instance with specified columns, architecture, metrics, optimizer, etc"

Initially my docstrings were much bigger/descriptive but Brian complained a lot about them as he said I was just describing the logic of the code and doc strings shouldn't be that.

I'm open to change this to anything you suggest.

Reply:
> describing the logic of the code and doc strings shouldn't be that.

Agreed about that.  The style guide says:

> A docstring should describe the function's calling syntax and its semantics, not its implementation.

The line between the two is pretty fine when the logic is simple.  What I'd wanted here was something about semantics -- what it meant to be configured.  I want to know what to expect from/about the returned object itself.  What's been done to it and what do I still need to do?  For example, does depth need to be normalized?

Reply:
changed the description hope you like it better.